### PR TITLE
ARROW-15195: [MATLAB] Enable GitHub Actions CI for MATLAB Interface on macOS

### PR DESCRIPTION
**Pull request by sreeharihegden**
## Overview
This Pull Request:

- Enables building of the MATLAB Interface C++ code, running of the C++ tests, and running of the MATLAB tests using GitHub Actions on a `macos-latest` VM.
- This is also updating the old version `v0` of [`matlab-actions/setup-matlab`](https://github.com/matlab-actions/setup-matlab) enabled for `ubuntu-latest` VM to the lastest `v1` [`matlab-actions/setup-matlab`](https://github.com/matlab-actions/setup-matlab).

## Implementation
This implementation uses [`matlab-actions`](https://github.com/matlab-actions) to automatically install MATLAB into a GitHub Actions Linux and Mac VM. 

We are using the most updated version: `v1` [`matlab-actions/setup-matlab`](https://github.com/matlab-actions/setup-matlab) to enable the CI for Mac. 

Here are the steps we require to build and run tests on Mac:
- Check out the repository
- Install ninja using the `brew` command
- Install MATLAB using [`matlab-actions/setup-matlab@v1`](https://github.com/matlab-actions/setup-matlab)
- Build MATLAB using the existing `matlab_build.sh` script in `ci/scripts`
- Set the `MATLABPATH` environment variable to the MATLAB Interface to Apache Arrow installation directory
- Run MATLAB tests in the folder `matlab/test` using [`matlab-actions/run-tests@v1`](https://github.com/matlab-actions/run-tests)

## Testing
1. The GitHub Actions workflows passed successfully with no errors in this [Job](https://github.com/mathworks/arrow/actions/runs/1796808254).
2. All C++ test results and MATLAB test results can be viewed in the workflow logs.

## Future Directions
Enable support for Windows: [`matlab-actions/setup-matlab@v1`](https://github.com/matlab-actions/setup-matlab) supports Windows now. We are working on enabling this for the MATLAB Arrow interface. 

## Notes

Thank you, Fiona @lafiona, for helping me with this pull request to submit the changes for `matlab.yml`.

### CODE REVIEW

**Comment by kou** [comment]
I've merged #12004. Could you rebase on master?

File: .github/workflows/matlab.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ concurrency:
 
 jobs:
 
-  matlab:
+  matlab-ubuntu:
```
Comment (kou) [inline .github/workflows/matlab.yml:40]: How about removing `matlab-` prefix because this file is only for MATLAB?

```suggestion
  ubuntu:
```

**Comment by kou** [comment]
Could you remove "Notes 1." from the description?

**Comment by sreeharihegden** [comment]
> Could you remove "Notes 1." from the description?

Thank you, @kou, for the feedback.
I have made the necessary changes to the job ids in `matlab.yml` as well as updated "Notes" here as suggested.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**